internal/builders/execaccessbuilder/create_access_resources.go

```diff
@@ -25,10 +25,18 @@ func (b *ExecAccessBuilder) CreateAccessResources(
 	// Cast the Template into an ExecAccessTemplate.
 	execTmpl := tmpl.(*v1alpha1.ExecAccessTemplate)
 
+	// If this resource already has a status.podName field set, then we respect
+	// that no matter what. We never mutate the pod that this access request
+	// was originally created for. Otherwise, pick a Pod and populate that
+	// status field.
+	if execReq.Status.PodName != "" {
+		return fmt.Sprintf("Pod already assigned -%s", execReq.GetName()), nil
```

Comment:
Wait .. now you're changing the access string right? By returning here, you are stoping the thread and actually removing the creation of the Role/RoleBindigns...

Reply:
I though the `SetPodName` gets set once we've created the role/rolebindings and prevents it from being updated again ?